File: old/session06_GrowArray_LinkedList_Stack_Queue/Queue/Queue.java
public class Queue<T> {
	private LinkedList2<T> impl;

	public Queue() { impl = new LinkedList1<>(); }
	public void enqueue(T val) { //O(1)
		impl.addEnd(val); //O(1)
	}
	
	public T dequeue() {
		return impl.removeStart();
	}
	
	public boolean isEmpty() {
		return impl.isEmpty();
	}
}

File: old/session06_GrowArray_LinkedList_Stack_Queue/Queue/Queue.cc
#include <iostream>
using namespace std;

class Queue {
private:
	LinkedList1 impl; // head only, one direction
public:
	void enqueue(int v) { // O(1)
		impl.addStart(v); //O(1)
	}
	int dequeue() { // O(n)
		return impl.removeEnd()
	}

	bool isEmpty() const { // 
    return impl.isEmpty();
	}
};
